### PR TITLE
upgrade docs build tools, typescript, and eslint

### PR DESCRIPTION
**Pull request by ben-polinsky**
Upgrade @itwin/build-tools to pre-release to take advantage of new typedoc version. TS bumped to 5.3.3. ESlint upgraded to 8.x. Upgrade @itwin/eslint-plugin to 4.0.x stable.

### CODE REVIEW

**Comment by ben-polinsky** [comment]
We need a fix from our eslint library for this to lint correctly: 

[PR](https://github.com/iTwin/eslint-plugin/pull/67) - [Commit](https://github.com/iTwin/eslint-plugin/pull/67/commits/485f27dbe9cc1852367e5b68cab0835377aa0d5c)

File: packages/browser/package.json [hunk 1]
```diff
@@ -35,9 +35,9 @@
     "oidc-client-ts": "^2.4.0"
   },
   "devDependencies": {
-    "@itwin/build-tools": "^4.0.0-dev.93",
+    "@itwin/build-tools": "^4.6.0-dev.27",
```
Comment (aruniverse) [inline packages/browser/package.json:38]: any reason to use pre-release?

File: packages/browser/src/types.ts [hunk 2]
```diff
@@ -44,14 +44,7 @@ export interface BrowserAuthorizationClientConfiguration
   /** A space-delimited collection of individual access claims specified by the authority. The user must consent to all specified scopes in order to grant authorization */
   readonly scope: string;
   /** The mechanism (or authentication flow) used to acquire auth information from the user through the authority */
-  readonly responseType?:
-  | "code"
```
Comment (aruniverse) [inline packages/browser/src/types.ts:48]: what was the reason for this?
wouldnt this make it harder for consumers to know what are actual valid response types?
not that the string union at the end helped too much

File: packages/oidc-signin-tool/src/SignInAutomation.ts [hunk 3]
```diff
@@ -26,7 +26,8 @@ interface AutomatedContextBase<T> {
   /** A function that takes the waitForCallback result data (e.g. a callback url)
    * and finalizes the sign in process
    */
-  resultFromCallback?: (t: T) => any | Promise<any>;
+  // eslint-disable-line @typescript-eslint/no-redundant-type-constituents
```
Comment (aruniverse) [inline packages/oidc-signin-tool/src/SignInAutomation.ts:29]: do you need both these disables?
should this be disable next line?

**Review by aruniverse — APPROVED** [review]

File: packages/browser/package.json (hunk 1 above)
Comment (ben-polinsky) [reply packages/browser/package.json:38]: Yes, we haven't yet published a version with new typedoc

File: packages/browser/src/types.ts (hunk 2 above)
Comment (ben-polinsky) [reply packages/browser/src/types.ts:48]: `@typescript-eslint/no-redundant-type-constituents : "code" is overridden by string in this union type.`
https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChkHIBGcUAXMgEQkBelyAPsgM5hSgDmuAvrrguhCtkMChiwBeHPkIlyyHkA - no autocomplete.

We could provide suggested valid values in the comment, or if we think we know all possible values, we could remove the `string`.

File: packages/oidc-signin-tool/src/SignInAutomation.ts (hunk 3 above)
Comment (ben-polinsky) [reply packages/oidc-signin-tool/src/SignInAutomation.ts]: At one point I thought I did, but re-tested and not needed. Updated.

File: packages/browser/src/types.ts (hunk 2 above)
Comment (ben-polinsky) [reply packages/browser/src/types.ts:48]: https://openid.net/specs/openid-connect-core-1_0.html#Authentication

Comment (aruniverse) [reply packages/browser/src/types.ts:48]: im not an auth expert so dont know all the other possible response types.
if this is the exhaustive list, then id rather remove the string

@evelynpreslar-bentley ?

Comment (aruniverse) [reply packages/browser/src/types.ts:48]: ah thx for sharing ben, looks like thats the list, and we can remove string

**Pull request merged**